docs/features/dynamic_tasks.md

```diff
@@ -0,0 +1,106 @@
+# Dynamic tasks
+
+DAG Factory supports Airflow’s
+[Dynamic Task Mapping](https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/dynamic-task-mapping.html),
+enabling workflows to dynamically create tasks at runtime.  This approach allows the number of tasks to be determined
+during execution, usually based on the outcome of a preceding task, rather than being predefined during DAG authoring.
+
```

Comment:
@pankajkoti do you think it would be worth to highlight some of the current limitations?

I tried to raise some of them as part of this PR description:
https://github.com/astronomer/dag-factory/pull/314

> - Assigning multiple parameters to a non-TaskFlow operator
> - Mapping over a task group
> - Transforming expanding data
> - Combining upstream data (aka “zipping”)

Reply:
Added these, thanks, please check.